### ISSUE TITLE
Block parsing for fenced code blocks with syntax highlighting in lists regression.

### ISSUE THREAD

**Issue by dmitshur** [russross/blackfriday#495]
I've just discovered PR #476 is causing the following regression in v1. /cc @tfogo @client9

Given this Markdown input:

<pre>
### Codeblock within list

-	list entry 1

	```
	if (i == 5)
	    break;
	```

-	list entry 2

### Stuff that comes later

stuff here
</pre>

Everything works as expected:

![image](https://user-images.githubusercontent.com/1924134/46589824-3e03ec00-ca7c-11e8-9ef9-e7510d013692.png)

However, if one uses a fenced code block with syntax highlighting, everything that comes after the fenced code block ends up being a part of "list entry 1":

<pre>
### Codeblock within list

-	list entry 1

	```C
	if (i == 5)
	    break;
	```

-	list entry 2

### Stuff that comes later

stuff here
</pre>

![image](https://user-images.githubusercontent.com/1924134/46589839-57a53380-ca7c-11e8-86aa-26145208ff57.png)

I.e., it parses _as if_ the input was:

<pre>
### Codeblock within list

-	list entry 1

	```C
	if (i == 5)
	    break;
	```

	-	list entry 2

	### Stuff that comes later

	stuff here
</pre>

**Comment by dmitshur** [comment]
I have a lead. Modifying the code as follows seems to fix the issue, at least for that one instance of the problematic input:

```diff
-_, marker := isFenceLine(chunk, nil, codeBlockMarker, false)
+_, marker := isFenceLine(chunk, new(string), codeBlockMarker, false)
```

I.e., passing a non-nil `info *string` parameter to `isFenceLine`.

**Comment by SamWhited** [comment]
I came here to report a similar issue which sounds like it might have the same cause, right now if a fenced code block immediately follows a list even if it's not indented it becomes part of the list. The following patch adds a test that can be used to demonstrate the behavior (the expected result might have wrong newlines, I just guessed):

```
diff --git a/block_test.go b/block_test.go
index 326c3110ce70..4db7d24156e2 100644
--- a/block_test.go
+++ b/block_test.go
@@ -1135,6 +1135,9 @@ func TestFencedCodeBlock(t *testing.T) {

                "```\n[]:()\n[]:)\n[]:(\n[]:x\n[]:testing\n[:testing\n\n[]:\nlinebreak\n[]()\n\n[]:\n[]()\n```",
                "<pre><code>[]:()\n[]:)\n[]:(\n[]:x\n[]:testing\n[:testing\n\n[]:\nlinebreak\n[]()\n\n[]:\n[]()\n</code></pre>\n",                                                                                                                      
+
+               "- test\n\n\n``` go\nfunc foo() bool {\n\treturn true;\n}\n```\n",
+               "<ul>\n<li>test</li>\n<ul>\n\n<pre><code class=\"language-go\">func foo() bool {\n\treturn true;\n}\n</code></pre>\n",                                                                                                                  
        }
        doTestsBlock(t, tests, FencedCode)
 }

```

**Comment by TeaWeb** [comment]
+1 the same issue here

**Comment by dmitshur** [comment]
Thank you very much for resolving this issue, @aignas and @rtfb!

You've also indirectly resolved shurcooL/markdownfmt#44.

**Comment by simonasGit** [comment]
#715 Same issue on V2
